lib/spack/spack/report.py

```diff
@@ -36,7 +36,8 @@
 import spack.package
 
 templates = {
-    'junit': os.path.join('reports', 'junit.xml')
+    'junit': os.path.join('reports', 'junit.xml'),
+    'cdash': os.path.join('reports', 'cdash')
```

Comment:
This probably needs to be changed as `templates` is not a list of templates anymore. An option would be to map keys to callable objects that write the report. Doing so we could move all the CDash specific code that is now in `collect_info` into a `CDashWriter` class that gets called only at `collect_info.__exit__`. Does it seem sensible?